diff --git a/arch/arm/dts/bcm2711-rpi-4-b.dts b/arch/arm/dts/bcm2711-rpi-4-b.dts
index 72ce80fb..eecbc8f2 100644
--- a/arch/arm/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/dts/bcm2711-rpi-4-b.dts
@@ -10,7 +10,7 @@
 
 	chosen {
 		/* 8250 auxiliary UART instead of pl011 */
-		stdout-path = "serial1:115200n8";
+		//stdout-path = "serial1:115200n8";
 	};
 
 	leds {
@@ -233,25 +233,36 @@
 };
 
 /* uart0 communicates with the BT module */
-&uart0 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32>;
-	uart-has-rtscts;
-	status = "okay";
-
-	bluetooth {
-		compatible = "brcm,bcm43438-bt";
-		max-speed = <2000000>;
-		shutdown-gpios = <&expgpio 0 GPIO_ACTIVE_HIGH>;
-	};
-};
+//&uart0 {
+//	pinctrl-names = "default";
+//	pinctrl-0 = <&uart0_ctsrts_gpio30 &uart0_gpio32>;
+//	uart-has-rtscts;
+//	status = "okay";
+//
+//	bluetooth {
+//		compatible = "brcm,bcm43438-bt";
+//		max-speed = <2000000>;
+//		shutdown-gpios = <&expgpio 0 GPIO_ACTIVE_HIGH>;
+//	};
+//};
 
 /* uart1 is mapped to the pin header */
-&uart1 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&uart1_gpio14>;
-	status = "okay";
-};
+//&uart1 {
+//	pinctrl-names = "default";
+//	pinctrl-0 = <&uart1_gpio14>;
+//	status = "okay";
+//};
+
+//&uart4 {
+//	u-boot,dm-pre-reloc;
+//	pinctrl-name = "default";
+//	pinctrl-0 = <&uart4_gpio8>;
+//	status = "okay";
+//};
+//
+//&uart4_gpio8 {
+//	u-boot,dm-pre-reloc;
+//};
 
 &vc4 {
 	status = "okay";
diff --git a/arch/arm/dts/bcm283x-u-boot.dtsi b/arch/arm/dts/bcm283x-u-boot.dtsi
index 22c67c42..297a23b3 100644
--- a/arch/arm/dts/bcm283x-u-boot.dtsi
+++ b/arch/arm/dts/bcm283x-u-boot.dtsi
@@ -25,13 +25,13 @@
 	};
 };
 
-&uart0 {
-	skip-init;
-	u-boot,dm-pre-reloc;
-};
+//&uart0 {
+//	//skip-init;
+//	u-boot,dm-pre-reloc;
+//};
 
 &uart1 {
-	skip-init;
+	//skip-init;
 	u-boot,dm-pre-reloc;
 };
 
@@ -39,9 +39,9 @@
 	u-boot,dm-pre-reloc;
 };
 
-&uart0_gpio14 {
-	u-boot,dm-pre-reloc;
-};
+//&uart0_gpio14 {
+//	u-boot,dm-pre-reloc;
+//};
 
 &uart1_gpio14 {
 	u-boot,dm-pre-reloc;
